### PR TITLE
DAOS-11811 control: Tighten up permissions of dRPC socket files

### PR DESCRIPTION
**Pull request by tanabarr**
/var/daos/daos_server/daos_server.sock has permissions 777, should be 600
/var/daos/daos_server/daos_engine_^pid^.sock has permission 755, should be 600
/var/daos/daos_agent/daos_agent.sock has permissions 777, should be 666

Required-githooks: true

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Review by Michael-Hennecke — APPROVED** [review]

**Comment by Michael-Hennecke** [comment]
also fix engine and agent socket permissions?

**Review by kjacque — DISMISSED** [review]
Very minor suggestions, not necessary for landing.

File: src/control/drpc/drpc_server_test.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ import (
 	"github.com/daos-stack/daos/src/control/logging"
 )
 
+var testFileMode os.FileMode = 0666
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/drpc/drpc_server_test.go:23]: change made, done

File: src/control/drpc/drpc_server.go [hunk 2]
```diff
@@ -98,9 +99,9 @@ func (d *DomainSocketServer) Start(ctx context.Context) error {
 	}
 	d.listener = lis
 
-	// TODO: Should we set more granular permissions? The only writer should
-	// be the I/O Engine and we should know which user is running it.
-	if err := os.Chmod(d.sockFile, 0777); err != nil {
+	// The only writer should be the I/O Engines which should be running as the same user as
```
Comment (knard-intel) [inline src/control/drpc/drpc_server.go:102]: Please, could you confirm that it is not needed to change the owner of the socket if the daos server is directly launch with the root account or systemd ?
From my understanding, should not be.

**Review by knard-intel — APPROVED** [review]
LGTM, just one existential question.

**Review by kjacque — APPROVED** [review]

**Comment by tanabarr** [comment]
Only CI failure is https://daosio.atlassian.net/browse/DAOS-12013 and unrelated to this change.

**Pull request merged**